src/OpenTelemetry.Instrumentation.AWS/Implementation/AWSTracingPipelineHandler.cs

```diff
@@ -172,6 +177,82 @@ private static void AddRequestSpecificInformation(Activity activity, IRequestCon
             SnsRequestContextHelper.AddAttributes(
                 requestContext, AWSMessagingUtils.InjectIntoDictionary(new PropagationContext(activity.Context, Baggage.Current)));
         }
+        else if (AWSServiceType.IsBedrockRuntimeService(service))
+        {
+            activity.SetTag(AWSSemanticConventions.AttributeGenAiSystem, "aws_bedrock");
+        }
+        else if (AWSServiceType.IsBedrockAgentService(service))
+        {
+            try
+            {
+                if (AWSServiceHelper.OperationNameToResourceMap().TryGetValue(AWSServiceHelper.GetAWSOperationName(requestContext), out var parameter))
+                {
+                    var property = request.GetType().GetProperty(parameter);
+                    if (property != null)
+                    {
+                        if (AWSServiceHelper.ParameterAttributeMap.TryGetValue(parameter, out var attribute))
+                        {
+                            activity.SetTag(attribute, property.GetValue(request));
+                        }
+                    }
+                }
+            }
+            catch (Exception)
+            {
+                // Guard against any reflection-related exceptions when running in AoT.
+                // See https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/1543#issuecomment-1907667722.
+            }
+        }
+    }
+
+    private static void AddResponseSpecificInformation(Activity activity, IResponseContext responseContext, string service)
+    {
+        AmazonWebServiceResponse response = responseContext.Response;
+
+        if (AWSServiceType.IsBedrockService(service))
```

Comment:
Can you use ServiceResponseParameterMap here as mentioned in my first comment: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1979/files#r1718974115